### PR TITLE
DOC: add vstack, hstack, dstack reference to stack documentation.

### PR DESCRIPTION
**Pull request by aukejw**
See issue #7183.

### CODE REVIEW

File: numpy/core/shape_base.py [hunk 1]
```diff
@@ -304,6 +304,11 @@ def stack(arrays, axis=0):
     See Also
     --------
     concatenate : Join a sequence of arrays along an existing axis.
+    vstack : Stack arrays in sequence vertically, along the existing row axis.
```
Comment (madphysicist) [inline numpy/core/shape_base.py]: "in sequence" is superfluous; "the existing" should be "an existing". Same applies to the other three additions.

File: numpy/core/shape_base.py [hunk 2]
```diff
@@ -304,6 +304,11 @@ def stack(arrays, axis=0):
...
+    hstack : Stack arrays in sequence horizontally, along the existing
+             column axis.
+    dstack : Stack arrays in sequence depth wise, along the existing
```
Comment (madphysicist) [inline numpy/core/shape_base.py]: I am pretty sure that "depthwise" is one word in this context.

Comment (aukejw) [inline numpy/core/shape_base.py]: I agree, should I change that in [`vstack`](https://github.com/numpy/numpy/blob/master/numpy/core/shape_base.py#L202), [`hstack`](https://github.com/numpy/numpy/blob/master/numpy/core/shape_base.py#L253) as well then?

File: numpy/core/shape_base.py (hunk 1 above)
Comment (aukejw) [inline numpy/core/shape_base.py]: You're right about 'in sequence', I'll remove it. There is only one 'row axis' though?

File: numpy/core/shape_base.py (hunk 2 above)
Comment (madphysicist) [inline numpy/core/shape_base.py]: For consistency, I would say yes, unless someone really likes it the old way and disagrees.

**Comment by madphysicist** [comment]
To answer your earlier question:

    >>> a = np.ones((2, 3, 4, 5, 6))
    >>> b = np.ones((2, 3, 4, 5, 6))
    >>> np.vstack((a, b)).shape
    (4, 3, 4, 5, 6)
    >>> np.hstack((a, b)).shape
    (2, 6, 4, 5, 6)
    >>> np.dstack((a, b)).shape
    (2, 3, 8, 5, 6)

Row is always first axis, column is always second axis and depth is always third, no matter how many axes the array has.

**Comment by aukejw** [comment]
Exactly, that's why I specified the axis as 'the' row axis, rather than 'an' axis. I'm no expert on the usage of definite vs indefinite articles though.. Thanks for the clarification!

**Comment by madphysicist** [comment]
In that case you were probably correct. Sorry for making you change that.

**Comment by madphysicist** [comment]
Perhaps you could say something like "... the existing row (first) axis ..."?

**Comment by shoyer** [comment]
I commented in the original issue, but just be to clear I'm -1 on this change until it's resolved.

**Comment by charris** [comment]
Thanks @aukejw .

**Comment by shoyer** [comment]
@charris I just said I'm -1 on this change! Can we revert?

**Pull request merged**